### PR TITLE
MIgrate to GH actions

### PR DESCRIPTION
**Pull request by adam-fowler**
Migrate CI to use GitHub Actions.

### Motivation:

To migrate to GitHub actions and centralised infrastructure.

### Modifications:

Changes of note:

Adopt swiftlang soundness checks, swift version matrix
Remove scripts and docker files which are no longer needed

### CODE REVIEW

**Comment by adam-fowler** [comment]
@rnro this is pretty much a copy of the swift-openapi-async-http-client setup with the license header check disabled as it adds Apple to the license headers.

**Comment by rnro** [comment]
Awesome. Looks good except for the `Package.swift` formatting failure. I'll have a look at if it would be possible to add a license header check for non-Apple repositories.

**Comment by adam-fowler** [comment]
> Awesome. Looks good except for the `Package.swift` formatting failure. 

Fixed

> I'll have a look at if it would be possible to add a license header check for non-Apple repositories.

Yeah would be good

**Comment by rnro** [comment]
Actually @adam-fowler you should be able to do something like this https://github.com/grpc/grpc-swift/blob/main/.license_header_template

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      linux_5_9_arguments_override: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error -Xswiftc -strict-concurrency=complete"
+      # TODO: `enable -Xswiftc -strict-concurrency=complete`
+      linux_5_10_arguments_override: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+      linux_6_0_arguments_override: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error -Xswiftc -require-explicit-sendable -Xswiftc -strict-concurrency=complete"
+      linux_nightly_6_0_arguments_override: "--explicit-target-dependency-import-check error -Xswiftc -require-explicit-sendable"
+      linux_nightly_main_arguments_override: "--explicit-target-dependency-import-check error -Xswiftc -require-explicit-sendable"
+
+  cxx-interop:
+    name: Cxx interop
+    uses: apple/swift-nio/.github/workflows/cxx_interop.yml@main
```
Comment (Joannis) [inline .github/workflows/pull_request.yml:28]: What's this used for?

**Review by Joannis — APPROVED** [review]

**Pull request merged**